CounterApp/App/AppCore.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  AppCore.swift
+//  CounterApp
+//
+//  Created by Wenjuan Li on 2022/7/12.
+//
+
+import ComposableArchitecture
+import Foundation
+
+struct AppState: Equatable {
+    var counter: CounterState
+    var lock: LockState
+}
+
+enum AppAction: Equatable {
+    case counter(CounterAction)
+    case lock(LockAction)
+}
+
+struct AppEnviroment {}
+
+let appReducer = Reducer<AppState, AppAction, AppEnviroment>.combine(
+    counterReducer.pullback(
+        state: \AppState.counter,
+        action: /AppAction.counter,
+        environment: { _ in CounterEnviroment.defaultEnv() }
+    ),
+    lockReducer.pullback(
+        state: \AppState.lock,
+        action: /AppAction.lock,
+        environment: { _ in LockEnvironment.defaultEnv() }
+    ),
+    Reducer { _, action, _ in
+        switch action {
+        default:
+            return .none
+        }
+    }
+).debug()
+
+let appStore = Store(
+    initialState: AppState(
+        counter: CounterState(),
+        lock: LockState(
+            counters: [
+                CounterState(count: 0),
+                CounterState(count: 5),
+                CounterState(count: 9)
+            ]
+        )
+    ),
+    reducer: appReducer,
+    environment: AppEnviroment()
+)
```

Comment:
One thing to highlight here is that this potentially gives access to the App Store from wherever part of the code, which may give more freedom and may cause incorrect implementations. Usually, we want to start off with the most global store and have that at our root view, so that's the only one having access to it. Then you would scope it down to more local domain stores as needed, but in order to guarantee the link between them, you just want to only make the parent available.

Reply:
@nanocba Thanks! Fixed.